File: 0263-ugly-number/0263-ugly-number.java
class Solution {
  public boolean isUgly(int n) {
    if (n <= 0) return false;
    if (n < 7) return true;
    int[] factors = new int[]{2, 3, 5};
    for (int f : factors) {
      while (n % f == 0) n /= f;
    }
    return n == 1;
  }
}

File: 0263-ugly-number/0263-ugly-number.py
class Solution:
  def isUgly(self, n: int) -> bool:
    if n <= 0: return False
    if n < 7: return True
    factors = [2, 3, 5]
    for f in factors:
      while n%f == 0:
        n //= f
    return n == 1